vcsclient/gitlab.go

```diff
@@ -447,19 +447,19 @@ func (client *GitLabClient) GetRepositoryEnvironmentInfo(_ context.Context, _, _
 // DownloadFileFromRepo on GitLab
 func (client *GitLabClient) DownloadFileFromRepo(_ context.Context, owner, repository, branch, path string) ([]byte, int, error) {
 	file, response, err := client.glClient.RepositoryFiles.GetFile(getProjectID(owner, repository), path, &gitlab.GetFileOptions{Ref: &branch})
-	if response != nil && response.StatusCode != http.StatusOK {
-		return nil, response.StatusCode, fmt.Errorf("expected %d status code while received %d status code with error:\n%s", http.StatusOK, response.StatusCode, err)
+	var statusCode int
+	if response != nil && response.Response != nil {
+		statusCode = response.Response.StatusCode
 	}
-	if err != nil {
-		return nil, 0, err
+	if statusCode != http.StatusOK {
+		return nil, statusCode, fmt.Errorf("expected %d status code while received %d status code with error:\n%s", http.StatusOK, response.StatusCode, err)
 	}
-
-	content, err := base64.StdEncoding.DecodeString(file.Content)
 	if err != nil {
-		return nil, response.StatusCode, err
+		return nil, statusCode, err
 	}
 
-	return content, response.StatusCode, err
+	content, err := base64.StdEncoding.DecodeString(file.Content)
+	return content, response.Response.StatusCode, err
```

Comment:
who said response is not nil? how said response.response is not nil?

Reply:
done